tests/test_pyoidc_facade.py

```diff
@@ -253,7 +254,7 @@ def test_client_credentials_grant(self):
         responses.add(responses.POST, token_endpoint,
                       json=client_credentials_grant_response)
         assert client_credentials_grant_response == facade.client_credentials_grant(
-            scope=['read', 'write'], audience=['client_id1, client_id2']).to_dict()
+            scope=scope, audience=['client_id1, client_id2']).to_dict()
```

Comment:
Do we really need to test whether an optional argument is present in the token request? We then have to test for `**kwargs` also. Whether or not it's included, it doesn't change the flow.  `access_token` will be issued regardless.

Reply:
I would say it's worth it - otherwise the parametrization of the test doesn't make a difference if the change in behavior between the two cases are not tested.
And, the testing of `**kwargs` would be a nice bonus - that they are passed as-is in the request.

Reply:
If I don't add parameterization, the if statement never becomes `False` which reduces the branch coverage. It's not worth to test if python dictionary is updated or not because by design we know that this line of code will always update `request` dictionary with `scope` and `kwargs`.

```python
request = {
    'grant_type': 'client_credentials',
    **kwargs
}
if scope:
        request['scope'] = ' '.join(scope)
```

Reply:
Branch coverage becomes even more meaningful if the behavior is verified as well:
* In the case of the `scope` parameter, the test will pass just the same if the `if` -statement is removed, which means that part of the code is actually not covered by tests.
* Same for the **kwargs - yes, the Python dictionary implementation is surely working but if I removed that addition of  **kwargs in `request` it won't be detected by the tests.

Reply:
A new change is coming for `client_credentials_grant` in my next PR for which there is a slight change in its test cases. I have addressed test cases for optional arguments in the new PR. You can merge this one.

Reply:
Ok - thanks!